### PR TITLE
 ramips: add support for JDCloud RE-CP-02

### PR DESCRIPTION
**Pull request by shenghuang147**
Specs:

+ Soc: MediaTek MT7621AT
+ RAM: 512 MB (DDR3)
+ Flash: 16 MB (SPI NOR)
+ WiFi: MediaTek MT7905DAN, MediaTek MT7975DN
+ Ethernet: 1 WAN, 3 LAN (Gigabit)
+ Buttons: Reset, Joylink
+ LEDs: (red, blue, green), routed to one indicator in the top of the device
+ Power: DC 12V 1A tip positive
+ 1 TF Card Slot

The pins for the serial console are already labeled on the board J4 (V, R, T, G). Serial settings: 3.3V, 115200

MAC addresses:

|         | MAC               | Algorithm |
| ------- | ----------------- | --------- |
| label   | dc:d8:xx:xx:xx:01 | label     |
| LAN     | dc:d8:xx:xx:xx:01 | label     |
| WAN     | dc:d8:xx:xx:xx:02 | label+1   |
| WLAN 2g | dc:d8:xx:xx:xx:03 | label+2   |
| WLAN 5g | de:d8:xx:xx:xx:04 | label+3   |

Flash instructions via serial console:

1. rename the openwrt-ramips-mt7621-jdcloud_re-cp-02-squashfs-sysupgrade.bin to JDCOS.bin

2. start a TFTP server from IP address 192.168.68.10 and serve the image named JDCOS.bin

3. connect your device to the LAN port

4. Power up the router and press any key on the console to interrupt the boot process.

5. enter the following commands on the router console

    1. `setenv bootcount 6`

    2. `saveenv`

    3. `reset` 

        > NOTE: wait for the restart, it will automatically fetch the image named JDCOS.bin from the TFTP server and write it into the flash. After the writing is completed, the router will be automatically restarted.

Known issues:

+ Unable to recognize large-capacity TF card. see #14042 But the patch [issuecomment-1910769942](https://github.com/openwrt/openwrt/issues/14042#issuecomment-1910769942) works

### CODE REVIEW

File: target/linux/ramips/mt7621/base-files/etc/init.d/bootcount [hunk 1]
```diff
@@ -15,6 +15,9 @@ boot() {
 			$((0xff)) ]] || printf '\xff' | dd of=/dev/mtdblock3 \
 			count=1 bs=1 seek=$((0x20001))
 		;;
+	jdcloud,re-cp-02)
+		echo -e "bootcount 0\nbootlimit 5\nupgrade_available 1"  | /usr/sbin/fw_setenv -s -
```
Comment (maurerle) [inline target/linux/ramips/mt7621/base-files/etc/init.d/bootcount:19]: Where can I find more information about why setting bootlimit is needed?
I don't know too much about it

**Review by maurerle — COMMENTED** [review]
Could you please rebase your single commit onto master and make the CI commit check pass?

File: target/linux/ramips/mt7621/base-files/etc/init.d/bootcount (hunk 1 above)
Comment (jhswartz) [reply target/linux/ramips/mt7621/base-files/etc/init.d/bootcount:19]: https://github.com/u-boot/u-boot/blob/master/doc/README.bootcount

**Comment by hauke** [comment]
Please fix the formal issues.
Please rebase the PR and add a commit message. The content of the PR description should be fine as commit message.

**Comment by shenghuang147** [comment]
Done #15372

**Pull request closed without merging**